addons/sale/models/sale_order_line.py

```diff
@@ -1160,6 +1160,9 @@ def _prepare_invoice_line(self, **optional_values):
             'is_downpayment': self.is_downpayment,
         }
         self._set_analytic_distribution(res, **optional_values)
+        if self.is_downpayment:
+            downpay_accounts = self.invoice_lines.filtered(lambda x: x.is_downpayment).account_id
+            res['account_id'] = downpay_accounts[0].id if downpay_accounts else False
```

Comment:
Thanks for the feedback !

Before 17.1, account is based on down payment product, but this product was removed in 17.1 and the account is now based on other product in sale (https://github.com/odoo/odoo/pull/150352/files#diff-e073b7fa9d45d46ba8d7f011257b0e77e1f87bf47982abc63dd618ff05dddb1aR264-R266)

Chose this code structure to avoid to override an account set by `optional_values`  to False triggering a Traceback when first invoice is created
https://github.com/odoo/odoo/blob/f793e32428e6b583a457ae2d09b2ae67c17041ac/addons/sale/models/sale_order_line.py#L1166-L1167

I will check the idiom and PR description and update the commit accordingly

Reply:
Ok, fine! 👍🏻

Reply:
Updated to use the idiom (thanks for the tip :+1: ) + modified description